CHANGELOG.md

```diff
@@ -1,3 +1,122 @@
+### citus v12.0.0 (July 11, 2023) ###
+
+PR #6866 : Add support for schema-based-sharding via a GUC
+TODO: PLEASE SHORTEN THE NEXT LINE MANUALLY, IT SHOULD BE NO LONGER THAN 78 CHARS
+* Adds citus.enable_schema_based_sharding GUC that allows sharding the database based on schemas when enabled.
+
+* Adds `citus_schema_distribute` and `citus_schema_undistribute` UDFs for
+  conversion between regular and distributed schemas (#6933)
+
+* Adds support for creating distributed tables without shard key (#6867)
+
+* Enable logical planner for single-shard tables (#6950)
+
+* Add citus_schemas view (#6979)
+
+* Enabling citus_stat_tenants to support schema-based tenants. (#6936)
+
+* Support CREATE TABLE .. AS SELECT .. commands for tenant tables (#6998)
+
+* Disallow single shard table creation if replication factor > 1 (#7006)
+
+PR #7027 : Improve error/hint messages related to schema-based sharding
+
+PR #6971 : Fix citus_table_type column in citus_tables and citus_shards views for single shard tables
+
+PR #6965 : Disable some udfs for tenant tables
+
+PR #6963 : Single Shard Misc UDFs 2
+
+PR #6924 : Add Support for Single Shard Tables in update_distributed_table_colocation
+
+PR #6916 : Citus UDFs support for single shard tables
+
+PR #6908 : Prevent downgrades when there is a single-shard table in the cluster
+
+PR #6902 : Disable master_create_empty_shard udf for single shard tables
+
+* Drops PG13 Support (#7002, #7007)
+
+PR #6975 : Citus Revise tree-walk APIs to include context
+
+PR #6974 : Random warning fixes
+
+PR #6957 : Fix some gucs' initial and boot values, and flag combinations
```

Comment:
@naisila Should we collect those items under PG16beta support?

Unrelated question: Should we also add the PR https://github.com/citusdata/citus/pull/6391?

Reply:
I think yes. But we should make its description more user-friendly. What do you think? Is below change OK?

"Adjusts query's CoerceViaIO & RelabelType nodes that are improper for deparsing" => "Fixes a bug related to text casts"

Reply:
> Should we collect those items under PG16beta support?

No, actually all those items can be ignored in the changelog, they are internal changes. PRs also don't have a DESCRIPTION.

> "Adjusts query's CoerceViaIO & RelabelType nodes that are improper for deparsing" => "Fixes a bug related to type casts with text"

Agreed, much more understandable. Given it can happen with varchar as well, let's make it Fixes a bug related to type casts with text/varchar